File: backend/src/main/java/es/udc/paproject/backend/model/services/TeamService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Team;
import es.udc.paproject.backend.model.exceptions.DuplicateInstanceException;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.UsedTrainingException;

public interface TeamService {

    Team addTeam(Long userId, String teamName, String arenaName, String ownerName, String description) throws InstanceNotFoundException, DuplicateInstanceException;

    void addTeamToSeason(Long seasonId, Long teamId, Long userId) throws InstanceNotFoundException;

    Team findTeamById(Long userId,Long teamId) throws InstanceNotFoundException;

    Team findTeamByPlayer(Long userId,Long playerId) throws InstanceNotFoundException;

    Team findTeamByName(Long userId, String teamName) throws InstanceNotFoundException;

    List<Team> findAllTeams(Long userId) throws InstanceNotFoundException;

    List<Team> findTeamsByName(Long userId,  String teamName) throws InstanceNotFoundException;

    List<Team> findTeamsByArena(Long userId,  String arena) throws InstanceNotFoundException;

    List<Team> findTeamsByOwner(Long userId,  String owner) throws InstanceNotFoundException;

    List<Team> findTeamsToSeason(Long userId, Long seasonId) throws InstanceNotFoundException;

    void removeTeam(Long userId, Long teamId) throws InstanceNotFoundException;

    void removeTeamToSeason(Long teamId, Long seasonId) throws InstanceNotFoundException, UsedTrainingException;

    void clearSeasonTeamTable(Long userId) throws InstanceNotFoundException;

    Team updateTeam(Long userId, Long teamId, String teamName, String arenaName, String ownerName, String description) throws InstanceNotFoundException;
}

File: frontend/src/backend/teamService.js
import {config, appFetch} from './appFetch';

export const findAllTeams = (onSuccess) =>
    appFetch('/teams', config('GET'), onSuccess);

export const findTeamById = (id, onSuccess) =>
    appFetch(`/teams/${id}`, config('GET'), onSuccess);

export const findTeamByName = (name, onSuccess) =>{
    appFetch(`/teams/?name=${name}`, config('GET'), onSuccess);
}
export const findTeamsByName = (name, onSuccess) =>{
    appFetch(`/teams/name?name=${name}`, config('GET'), onSuccess);
}
export const findTeamsByArena = (arena, onSuccess, onErrors) =>{
    appFetch(`/teams/arena?arena=${arena}`, config('GET'), onSuccess, onErrors);
}
export const findTeamsByOwner = (owner, onSuccess) =>{
    appFetch(`/teams/owner?owner=${owner}`, config('GET'), onSuccess);
}
export const addTeam = (name, arenaName, ownerName, description, onSuccess, onErrors) =>
    appFetch(`/teams?name=${name}&arenaName=${arenaName}&ownerName=${ownerName}&description=${description}`, config('POST'), onSuccess, onErrors);

export const updateTeam = (id, name, onSuccess, onErrors) =>
    appFetch(`/teams/${id}?name=${name}`, config('PUT'), onSuccess, onErrors);

export const removeTeam = (id, onSuccess, onErrors) =>{
    appFetch(`/teams/${id}`, config('DELETE'), onSuccess, onErrors);
}

export const addTeamToSeason = (seasonId, teamId, onSuccess, onErrors) =>
    appFetch(`/teams/${seasonId}/addTeamtoSeason/?teamId=${teamId}`, config('POST'), onSuccess, onErrors);

export const findTeamsToSeason = (seasonId, onSuccess, onErrors) => 
    appFetch(`/teams/${seasonId}/season`, config('GET'), onSuccess, onErrors);

export const removeTeamToSeason = (seasonId, teamId, onSuccess, onErrors) =>
    appFetch(`/teams/${seasonId}/season?teamId=${teamId}`, config('DELETE'), onSuccess, onErrors);